File: spot-the-difference/__tests__/spotTheDifference.test.js
const { spot } = require('../spotTheDifference');

describe('Given two strings, return an array containing the indices of the different character', () => {
    test("Given 'abcdefg' & 'abcqetg'), return [3, 5]", () => {
        const string1 = 'abcdefg';
        const string2 = 'abcqetg';
        expect(spot(string1, string2)).toEqual([3, 5]);
    });
    test("Given 'Hello World!' & 'hello world.', return [0, 6, 11]", () => {
        const string1 = 'Hello World!';
        const string2 = 'hello world.';
        expect(spot(string1, string2)).toEqual([0, 6, 11]);
    });
    test("Given 'FixedGrey' & 'FixedGrey', return []", () => {
        const string1 = 'FixedGrey';
        const string2 = 'FixedGrey';
        expect(spot(string1, string2)).toEqual([]);
    });
    test("Given 'bD B' & 'BD B', return [0]", () => {
        const string1 = 'bD B';
        const string2 = 'BD B';
        expect(spot(string1, string2)).toEqual([0]);
    });
    test("Given 's/LrQ TiVXDlhnSeWnieGp' & 's/LrQ TiVXDlHnSeWNieGp', return [12, 17]", () => {
        const string1 = 's/LrQ TiVXDlhnSeWnieGp';
        const string2 = 's/LrQ TiVXDlHnSeWNieGp';
        expect(spot(string1, string2)).toEqual([12, 17]);
    });
    test("Given 'SXExDWf BBd id HU lZl s' & 'SXExDWf BBd ID HU LZl S', return [ 12, 13, 18, 22 ]", () => {
        const string1 = 'SXExDWf BBd id HU lZl s';
        const string2 = 'SXExDWf BBd ID HU LZl S';
        expect(spot(string1, string2)).toEqual([12, 13, 18, 22]);
    });
    test("Given 'G.L    Wj e .mpWozP  !Gh Z' & 'G.L    Wj e .mpWoZP  !Gh Z', return [ 17 ]", () => {
        const string1 = 'G.L    Wj e .mpWozP  !Gh Z';
        const string2 = 'G.L    Wj e .mpWoZP  !Gh Z';
        expect(spot(string1, string2)).toEqual([17]);
    });
    test("Given 's/LrQ TiVXDlhnSeWnieGp' & 's/LrQ TiVXDlHnSeWNieGp', return [12, 17]", () => {
        const string1 = 's/LrQ TiVXDlhnSeWnieGp';
        const string2 = 's/LrQ TiVXDlHnSeWNieGp';
        expect(spot(string1, string2)).toEqual([12, 17]);
    });
});



//  bD B  BD B  [ 0 ]

//https://www.codewars.com/kata/5881460c780e0dd207000084/train/javascript

File: python-venv/spot_the_dffierence/__tests__/spot_the_difference_test.py
from kata_function.spot_the_difference import spot

def test_1():
    string1 = 'abcdefg'
    string2 = 'abcqetg'
    assert spot(string1, string2) == [3, 5]

def test_2():
    string1 = 'Hello World!'
    string2 = 'hello world.'
    assert spot(string1, string2) == [0, 6, 11]

def test_3():
    string1 = 'FixedGrey'
    string2 = 'FixedGrey'
    assert spot(string1, string2) == []

def test_4():
    string1 = 'bD B'
    string2 = 'BD B'
    assert spot(string1, string2) == [0]

def test_5():
    string1 = 's/LrQ TiVXDlhnSeWnieGp'
    string2 = 's/LrQ TiVXDlHnSeWNieGp'
    assert spot(string1, string2) == [12, 17]

def test_6():
    string1 = 'SXExDWf BBd id HU lZl s'
    string2 = 'SXExDWf BBd ID HU LZl S'
    assert spot(string1, string2) == [12, 13, 18, 22]

def test_7():
    string1 = 'G.L    Wj e .mpWozP  !Gh Z'
    string2 = 'G.L    Wj e .mpWoZP  !Gh Z'
    assert spot(string1, string2) == [17]

#https://www.codewars.com/kata/5881460c780e0dd207000084/train/python